File: NotesAppRoomJava/app/src/main/java/com/example/notesapproom_java/view/EditNoteFragment.java
package com.example.notesapproom_java.view;

import android.app.AlertDialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.view.MenuHost;
import androidx.core.view.MenuProvider;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Lifecycle;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;

import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.example.notesapproom_java.MainActivity;
import com.example.notesapproom_java.R;
import com.example.notesapproom_java.databinding.FragmentEditNoteBinding;
import com.example.notesapproom_java.model.Note;
import com.example.notesapproom_java.viewmodel.NoteViewModel;


public class EditNoteFragment extends Fragment implements MenuProvider {

    private FragmentEditNoteBinding binding;
    private NoteViewModel notesViewModel;
    private Note currentNote;

    public EditNoteFragment() {
        super(R.layout.fragment_edit_note);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentEditNoteBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        MenuHost menuHost = requireActivity();
        menuHost.addMenuProvider(this, getViewLifecycleOwner(), Lifecycle.State.RESUMED);

        notesViewModel = ((MainActivity) requireActivity()).getNoteViewModel();

        if (getArguments() != null) {
            EditNoteFragmentArgs args = EditNoteFragmentArgs.fromBundle(getArguments());
            currentNote = args.getNote();
        }

        binding.editNoteTitle.setText(currentNote.getNoteTitle());
        binding.editNoteDesc.setText(currentNote.getNoteDesc());

        binding.editNoteFab.setOnClickListener(v -> {
            String noteTitle = binding.editNoteTitle.getText().toString().trim();
            String noteDesc = binding.editNoteDesc.getText().toString().trim();

            if (!noteTitle.isEmpty()) {
                Note updatedNote = new Note(currentNote.getId(), noteTitle, noteDesc);
                notesViewModel.updateNote(updatedNote);
                Navigation.findNavController(v).popBackStack(R.id.homeFragment, false);
            } else {
                Toast.makeText(getContext(), "Please enter note title", Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void deleteNote() {
        new AlertDialog.Builder(getActivity())
                .setTitle("Delete note")
                .setMessage("Do you want to delete this note?")
                .setPositiveButton("Delete", (dialog, which) -> {
                    notesViewModel.deleteNote(currentNote);
                    Toast.makeText(getContext(), "Note Deleted", Toast.LENGTH_SHORT).show();
                    NavController navController = NavHostFragment.findNavController(this);
                    navController.popBackStack(R.id.homeFragment, false);
                })
                .setNegativeButton("Cancel", null)
                .create()
                .show();
    }

    @Override
    public void onCreateMenu(@NonNull Menu menu, @NonNull MenuInflater menuInflater) {
        menu.clear();
        menuInflater.inflate(R.menu.menu_delete, menu);
    }

    @Override
    public boolean onMenuItemSelected(@NonNull MenuItem menuItem) {
        if (menuItem.getItemId() == R.id.deleteMenu) {
            deleteNote();
            return true;
        }
        return false;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        binding = null;
    }
}

File: NotesAppRoom/app/src/main/java/com/example/notesapproom/fragments/EditNoteFragment.kt
package com.example.notesapproom.fragments

import android.app.AlertDialog
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.view.MenuHost
import androidx.core.view.MenuProvider
import androidx.lifecycle.Lifecycle
import androidx.navigation.findNavController
import androidx.navigation.fragment.navArgs
import com.example.notesapproom.MainActivity
import com.example.notesapproom.R
import com.example.notesapproom.databinding.FragmentEditNoteBinding
import com.example.notesapproom.model.Note
import com.example.notesapproom.viewmodel.NoteViewModel


class EditNoteFragment : Fragment(R.layout.fragment_edit_note), MenuProvider {

    private var editNoteBinding: FragmentEditNoteBinding? =null
    private val binding get() = editNoteBinding!!

    private lateinit var notesViewModel: NoteViewModel
    private lateinit var currentNote : Note

    private val args: EditNoteFragmentArgs by navArgs()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        editNoteBinding = FragmentEditNoteBinding.inflate(inflater, container, false)

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val menuHost: MenuHost = requireActivity()
        menuHost.addMenuProvider(this, viewLifecycleOwner, Lifecycle.State.RESUMED)

        notesViewModel = (activity as MainActivity).noteViewModel
        currentNote = args.note!!

        binding.editNoteTitle.setText(currentNote.noteTitle)
        binding.editNoteDesc.setText(currentNote.noteDesc)

        binding.editNoteFab.setOnClickListener {
            val noteTitle = binding.editNoteTitle.text.toString().trim()
            val noteDesc = binding.editNoteDesc.text.toString().trim()

            if (noteTitle.isNotEmpty()) {
                val note = Note(currentNote.id, noteTitle, noteDesc)
                notesViewModel.updateNote(note)
                view.findNavController().popBackStack(R.id.homeFragment, false)
            } else {
                Toast.makeText(context, "Please enter note title", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun deleteNote(){
        AlertDialog.Builder(activity).apply {
            setTitle("Delete note")
            setMessage("Do you want to delete this note?")
            setPositiveButton("Delete") {_,_ ->
                notesViewModel.deleteNote(currentNote)
                Toast.makeText(context, "Note Deleted", Toast.LENGTH_SHORT).show()
                view?.findNavController()?.popBackStack(R.id.homeFragment, false)
            }
            setNegativeButton("Cancel", null)
        }.create().show()
    }

    override fun onCreateMenu(menu: Menu, menuInflater: MenuInflater) {
        menu.clear()
        menuInflater.inflate(R.menu.menu_delete, menu)
    }

    override fun onMenuItemSelected(menuItem: MenuItem): Boolean {
        return when(menuItem.itemId){
            R.id.deleteMenu -> {
                deleteNote()
                true
            } else -> false
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        editNoteBinding!=null
    }

}